### PR TITLE
Restless change to POST for the HTTP stack.

### PR DESCRIPTION
**Pull request by tanertopal**
#### What does this implement/fix? Explain your changes.

Change GET tasks to POST so that we can in the future include data in the GetTaskRequest without leaking anything through the URL

### CODE REVIEW

File: src/py/flwr/server/rest_server/rest_api.py [hunk 1]
```diff
@@ -42,8 +42,19 @@
...
 
-@app.get("/api/1.1/tasks/", response_class=Response)
-def tasks(response: Response, client_id: Optional[int] = None) -> Response:
+@app.post("/api/1.1/tasks", response_class=Response)
+async def tasks(request: Request) -> Response:
+    # This is required to get the request body as raw bytes
+    get_tasks_req_msg_bytes: bytes = await request.body()
+
+    # Deserialize ProtoBuf
+    get_tasks_req_msg = CreateResultsRequest()
```
Comment (danieljanes) [inline src/py/flwr/server/rest_server/rest_api.py:51]: Shouldn't this be `GetTasksRequest`?

File: src/py/flwr/server/rest_server/rest_api.py [hunk 2]
```diff
@@ -52,7 +63,7 @@ def tasks(response: Response, client_id: Optional[int] = None) -> Response:
     task_resp_msg = GetTasksResponse()
     task_resp_msg.tokenized_tasks.tokenized_tasks.append(tokenized_task)
     task_resp_bytes = task_resp_msg.SerializeToString()
-    print(f"GET - Sending Task {message_being_sent}:")
+    print(f"POST - Sending GetTaskResponse {message_being_sent}:")
```
Comment (danieljanes) [inline src/py/flwr/server/rest_server/rest_api.py:66]:
```suggestion
    print(f"POST - Sending GetTasksResponse {message_being_sent}:")
```

File: src/py/flwr/server/rest_server/rest_api.py [hunk 3]
```diff
@@ -61,7 +72,7 @@ def tasks(response: Response, client_id: Optional[int] = None) -> Response:
     )
 
 
-@app.post("/api/1.1/result/")
+@app.post("/api/1.1/result")
 async def result(request: Request) -> Response:  # Check if token is needed here
```
Comment (danieljanes) [inline src/py/flwr/server/rest_server/rest_api.py:76]: The request can contain multiple results, should we make this plural?

```suggestion
async def results(request: Request) -> Response:  # Check if token is needed here
```

File: src/py/flwr/client/rest_client/connection.py [hunk 4]
```diff
@@ -62,13 +62,17 @@ def rest_not_a_connection(
 
     def receive() -> Optional[ServerMessage]:
         """Receive next task from server."""
+       # Serialize ProtoBuf to bytes
```
Comment (danieljanes) [inline src/py/flwr/client/rest_client/connection.py:65]:
```suggestion
        # Serialize ProtoBuf to bytes
```

**Review by danieljanes — CHANGES_REQUESTED** [review]

File: src/py/flwr/server/rest_server/rest_api.py (hunk 1 above)
Comment (tanertopal) [reply src/py/flwr/server/rest_server/rest_api.py]: You are correct. Fixed it.

File: src/py/flwr/server/rest_server/rest_api.py [hunk 5]
```diff
@@ -61,8 +73,8 @@ def tasks(response: Response, client_id: Optional[int] = None) -> Response:
     )
 
 
-@app.post("/api/1.1/result/")
-async def result(request: Request) -> Response:  # Check if token is needed here
+@app.post("/api/1.1/result")
```
Comment (danieljanes) [inline src/py/flwr/server/rest_server/rest_api.py:76]:
```suggestion
@app.post("/api/1.1/results")
```

File: src/py/flwr/client/rest_client/connection.py [hunk 6]
```diff
@@ -99,11 +103,11 @@ def send(client_message: ClientMessage) -> None:
 
         # Send ClientMessage to server
         r = requests.post(
-            f"{base_url}/api/1.1/result/",
+            f"{base_url}/api/1.1/result",
```
Comment (danieljanes) [inline src/py/flwr/client/rest_client/connection.py:106]:
```suggestion
            f"{base_url}/api/1.1/results",
```

File: src/py/flwr/client/rest_client/connection.py [hunk 7]
```diff
@@ -99,11 +103,11 @@ def send(client_message: ClientMessage) -> None:
...
             headers={"Content-Type": "application/protobuf"},
             data=results_req_msg_bytes,
         )
-        print(f"[C-{client_id}] POST /api/1.1/result/:", r.status_code, r.headers)
+        print(f"[C-{client_id}] POST /api/1.1/result:", r.status_code, r.headers)
```
Comment (danieljanes) [inline src/py/flwr/client/rest_client/connection.py:110]:
```suggestion
        print(f"[C-{client_id}] POST /api/1.1/results:", r.status_code, r.headers)
```

**Review by danieljanes — CHANGES_REQUESTED** [review]

**Pull request merged**